diff --git a/bindings/yarp.i b/bindings/yarp.i
--- a/bindings/yarp.i
+++ b/bindings/yarp.i
@@ -310,8 +310,8 @@ typedef unsigned char yarpPixelMono;
         return NULL;
     }
 
-    $1 = (int)PyInt_AsLong(PyTuple_GetItem($input,0));   /* int i */
-    $2 = (int)PyInt_AsLong(PyTuple_GetItem($input,1));   /* int j */
+    $1 = (int)PyLong_AsLong(PyTuple_GetItem($input,0));   /* int i */
+    $2 = (int)PyLong_AsLong(PyTuple_GetItem($input,1));   /* int j */
 };
 #endif
 
